### DISCUSSION TITLE
BlockManager issue.

### DISCUSSION THREAD

**Discussion by turgunov01** [GrapesJS/grapesjs#5935] [General]
Unable to create two blockManager panels to filter the blocks, into using and not-using blocks. 

Generally, I have to create two blocks(panels) which must be different, but config can not have two blockManager.appendTo keys, so i have to handle my blocks somehow for filtering them, using and not-using blocks.

I have an idea actually, to clip a key for a Objects inside of the Blocks, like after using sending request to API which is returning the blocks List, and each component:add or component:remove use key such: used: true/false

I am waiting for some ideas, and support with my question.

**Comment by glaprida** [comment]
I needed something similar (im my case it was Basic Blocks vs. Layouts) and instead of using blockManager.appendTo I did the following:

```
const filterBlocks = () =>{
  let allBlocks = editor.value.Blocks.getAll()
  
  let layouts = ["Layouts category 1", "Layouts category 2"];

  const filteredLayouts = allBlocks.filter(block => layouts.includes( block.get('category') ) );
  const filteredBlocks = allBlocks.filter(block => !layouts.includes( block.get('category') ) );

  const allBlocksEl = editor.value.Blocks.render(filteredBlocks, { external: true });
  const allLayoutsEl = editor.value.Blocks.render(filteredLayouts, { external: true });

  document.getElementById('blocks').appendChild(allBlocksEl);
  document.getElementById('layouts').appendChild(allLayoutsEl);
}

editor.on('load', (editor)=>{
        filterBlocks()
}
```

In my case I filter them by category, but you can filter by whatever you want.

**Reply by turgunov01** [reply]
Oh, I see, I had this version, but my templates for blockManager are being fetched from another server via Fetch API, so when i change category, it means i should refetch the content by types? Or should i fetch them all and filter?

**Reply by glaprida** [reply]
I would fetch them all on load and filter them, unless the fetch data is so big and so rarely used that it's not worth it. (like if you have 1000 layouts and its on a tab that only some users open)

**Reply by turgunov01** [reply]
Oh, I see. If I gotta filter them I think it will be easier to handle, but I was willing to make two blockManagers and as the result there is no any possibility to make it, so i think it will be easier to filter them to categories, but i somehow can not understand it.
